File: metarealm-growth-os-main/backend/app/models/outreach.py
"""Outreach drafts and generated proposals — the money team's output."""

from sqlalchemy import ForeignKey, String, Text
from sqlalchemy.orm import Mapped, mapped_column

from app.core.db import Base


class OutreachDraft(Base):
    __tablename__ = "outreach_drafts"

    id: Mapped[str] = mapped_column(String, primary_key=True)
    company_id: Mapped[str] = mapped_column(
        ForeignKey("companies.id"), nullable=False
    )
    company: Mapped[str] = mapped_column(String, nullable=False)
    contact_name: Mapped[str | None] = mapped_column(String, nullable=True)
    kind: Mapped[str] = mapped_column(String, nullable=False)  # intro | follow_up
    subject: Mapped[str] = mapped_column(String, nullable=False)
    body: Mapped[str] = mapped_column(Text, nullable=False)
    status: Mapped[str] = mapped_column(String, nullable=False, default="draft")
    created_at: Mapped[str] = mapped_column(String, nullable=False)


class Proposal(Base):
    __tablename__ = "proposals"

    id: Mapped[str] = mapped_column(String, primary_key=True)
    opportunity_id: Mapped[str] = mapped_column(
        ForeignKey("opportunities.id"), nullable=False
    )
    company: Mapped[str] = mapped_column(String, nullable=False)
    title: Mapped[str] = mapped_column(String, nullable=False)
    filename: Mapped[str] = mapped_column(String, nullable=False)
    created_at: Mapped[str] = mapped_column(String, nullable=False)

File: reference/metarealm-growth-os-main/frontend/types/outreach.ts
/** Outreach and proposals — mirrors backend/app/schemas/outreach.py. */

export interface OutreachDraft {
  id: string;
  companyId: string;
  company: string;
  contactName?: string | null;
  kind: "intro" | "follow_up";
  subject: string;
  body: string;
  status: "draft" | "sent";
  createdAt: string;
}

export interface Proposal {
  id: string;
  opportunityId: string;
  company: string;
  title: string;
  filename: string;
  createdAt: string;
}
